src/Microsoft.DotNet.Interactive.Http.Parsing/Parsing/HttpRequestParser.cs

```diff
@@ -131,6 +131,15 @@ public HttpSyntaxTree Parse()
                         node = new HttpVariableValueNode(_sourceText, _syntaxTree);
 
                         ParseLeadingWhitespaceAndComments(node);
+                    } 
+                    else if (IsAtStartOfEmbeddedExpression())
+                    {
+                        node = new HttpVariableValueNode(_sourceText, _syntaxTree);
+                        node.Add(ParseEmbeddedExpression());
+                        if(CurrentToken is { Kind: HttpTokenKind.NewLine })
```

Comment:
Is this check needed considering that the while loop condition above on line 125 already checks for this?

Reply:
This check is important because we only want to break after parsing the embedded expression if there's a newline because {{host}}/anything should be considered valid and we'll want to consume the slash and following tokens through the while loop

Reply:
Won't that happen automatically even without this condition and break though? Looks like if we don't add this condition and we are at a newline, we will simply continue to the while loop condition above and break out of the loop over there. Am I missing something?

Reply:
So it will actually be consumed into the node because the end of the is node null check it sees if there's a start of an embedded expression and if not it's consumed into the node